### ISSUE TITLE
Better approach for initialization and property changed behavior within the Actor Lifecycle

### ISSUE THREAD

**Issue by xuelongmu** [CesiumGS/cesium-unreal#385]
@nithinp7, @javagl and I have been discussing the UE4 actor lifecycle and when it makes the most sense to call certain methods. These are our goals:

1) Ensure that properties are properly initialized at editor-time, at run-time, and when transitioning from editor-time to PIE (play in editor). #362 
2) Respond to property updates at editor-time appropriately
3) Bind delegates to editor-time or run-time events (e.g. mouse click to select actors) #335 

With #381 @javagl was able to glean some useful information about the actor lifecycle and the potential issues of using the constructor or `OnConstruction` methods.

So far we have the following insights:
-  The constructor and `OnConstruction` methods can be called multiple times.
- `PostLoadProperties` seems to be called only once. (supported by [this thread](https://answers.unrealengine.com/questions/846470/difference-between-postinitproperties-and-postedit.html)). We can verify this by asking on UDN, and they may also know of a more suitable method. This is important because we want to handle goals 1) and 3) as early as we can in the actor's lifecycle, but without the side effects of calling these initialization methods multiple times.
- Responding to editor-time property changes should be handled in `PostEditChangeProperty` or `PostEditChangeChainProperty` (for ustructs or arrays). We can get the name of the changed property, and then choose how we respond to different property changes. It seems like certain property changes (e.g. for Cesium3DTileset) are handled in `OnConstruction` and we will want to move them out.
- Responding to run-time property changes should be handled in C++ `BlueprintCallable `setter methods

Please chime in if there's anything I haven't mentioned, I probably forgot a couple things as we've had several discussions/posts over the past week around this topic :)

The discussion of what should be stored as state (UPROPERTY) vs. what should be recomputed is best left for another discussion, I think.

**Comment by javagl** [comment]
One thing that isn't explicitly mentioned is the order of the initialization of objects. Maybe it's not critically important right now, maybe it's a separate issue, and maybe it's just about finding the right place in the documentation, and in any case, it's hard to articulate it precisely. 

But for an "object graph", say two tilesets that share one GeoRef instance, I wonder in which order the objects are "cloned" when switching from the Editor into PIE mode: Will it clone the tilesets first, and then assign the cloned GeoRef to both of them? And if so: Is there a "known place" where this happens? 

This may be of interest in general, but is specifically related to our current state, where each actor stores a GeoRef, but the GeoRef also stores references to the actors. I wonder whether Unreal can resolve these mutual references, or whether the reason why we didn't see any glitches there until now is just a result of calling `this->Georeference->AddGeoreferencedObject(this);` occasionally...

**Comment by kring** [comment]
I believe this was addressed in #430. Please reopen if I'm mistaken.
